src/liboslexec/backendllvm.cpp

```diff
@@ -755,8 +758,8 @@ llvm::Value*
 BackendLLVM::userdata_initialized_ref(int userdata_index)
 {
     int fieldnum = 1;  // field 1 is the userdata_initialized array
```

Comment:
Once difference here is that groupdata_field_ref created some additional name/debug info associated with the GEP instruction.  

```
llvm::Value*
BackendLLVM::groupdata_field_ref(int fieldnum)
{
    return ll.GEP(llvm_type_groupdata(), groupdata_ptr(), 0, fieldnum,
                  llnamefmt("{}_ref", m_groupdata_field_names[fieldnum]));
}
```

As pointers become typeless, the names/debug info we generate might become more and more important.  Perhaps a follow on task after this transition.  But for those of us who end up reading llvm-ir, any metadata left around at all can be very useful.

Reply:
I added (back) names to those groupdata GEP instructions now, both in batched and non-batched code.